### PR TITLE
856 more toolbox metal test cases

### PR DESCRIPTION
**Pull request by cdelnano**
### What are the issues this pull addresses (issue numbers / links)?
Issue #856 

### Did you add tests to cover your changes (yes/no)?
No

### Did you update the documentation accordingly (yes/no)?
Yes 

### Did you read the CONTRIBUTING document (yes/no)?
Yes

### Summary
Added unit testing for layer_stack_handler.py and bounds_for_path_and_poly_tables.py using the multiplanar design provided by @AbeerVaishnav13 .

### CODE REVIEW

**Comment by cdelnano** [comment]
Branch can be deleted after merge

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 1]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+
+    def test_toolbox_metal_get_bounds_of_path_and_poly_tables(self):
+        """Test functionality of get_bounds_of_path_and_poly_tables in toolbox_metal.py"""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:370]: I think this is chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 2]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+                                       pad_width="120um",
+                                       pad_height="30um",
+                                       cpw_stub_height="250um",
+                                       chip="c_chip",
+                                       layer=1))
+
+        self.assertIsInstance(
+            BoundsForPathAndPolyTables(
+                multiplanar_design).get_bounds_of_path_and_poly_tables(
+                    False, [], 1, 0, 0), tuple)
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:386]: There are many things returned in the tuple.  
Could you check the tuple that the total number of items has not changed, and possibly check the type?

 Returns:
            tuple[tuple[float, float, float, float], pd.DataFrame, pd.DataFrame, bool, Union[None,set]]:
                                    tuple: minx, miny, maxx, maxy values based
                                                    on either total chip size or box_plus_buffer.
                                                    In xy plane, the box_for_xy_bounds will be limited
                                                    by the chip_bounds_xy if box_for_xy_bounds is larger
                                                    than chip_bounds_xy.
                                    pd.DataFrame: The path and poly dataframes concatenated for qcomp_ids
                                    pd.DataFrame: The path, poly, junction dataframes concatenated for qcomp_ids
                                    bool: If there is a key in design with same chip_name as in layer_stack
                                    Union[None,set]: Either None if the names don't match,
                                                    or the set of chip_names that can be used.

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 3]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        for item in coords.values():
+            x_0 = item['center_x']
+            y_0 = item['center_y']
+            x_1 = item['size_x']
+            y_1 = item['size_y']
+        x_0 = int(float(x_0[:-2]))
+        y_0 = int(float(y_0[:-2]))
+        x_1 = int(x_1[:-2])
+        y_1 = int(y_1[:-2])
+        self.assertEqual(
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:420]: Nice!

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 4]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+                                       layer=1))
+        multiplanar_design.chips['c_chip'] = {
+            'size': {
+                'center_x': '0.0mm',
+                'center_y': '0.0mm',
+                'size_x': '9mm',
+                'size_y': '7mm'
+            }
+        }
+        multiplanar_design.chips['c_chip'] = {
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:457]: I think is a duplicate assignment.

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 5]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:498]: This looks like a duplicate.

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 6]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of ensure_component_box_smaller_than_chip_box in toolbox_metal.py"""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:398]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 7]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of get_box_for_xy_bounds in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:439]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 8]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of are_all_chipnames_in_design in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:480]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 9]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of get_x_y_for_chip in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:521]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 10]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of chip_names_not_in_design in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:564]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 11]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of chip_size_not_in_chipname_within_design in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:590]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 12]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of get_layer_datatype_when_fill_is_true in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:614]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 13]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of get_properties_for_layer_datatype in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:656]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 14]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of is_layer_data_unique in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:687]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 15]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of read_csv_df in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:712]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 16]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of get_unique_chip_names in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:737]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 17]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of get_unique_layer_ints in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:762]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 18]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        """Test functionality of _warning_properties in toolbox_metal.py."""
+        ls_file_path = ("./qiskit_metal/tests/test_data/planar_chip.txt")
+        multiplanar_design = MultiPlanar(metadata={},
+                                         overwrite_enabled=True,
+                                         layer_stack_filename=ls_file_path)
+        conn_pads = dict(connection_pads=dict(coup1=dict(loc_W=-1, loc_H=1),
+                                              coup2=dict(loc_W=1, loc_H=1)))
+        q1 = TransmonPocket6(multiplanar_design,
+                             "Q1",
+                             options=dict(**conn_pads, chip="q_chip", layer=2))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:787]: chip="c_chip", layer=1

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 19]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+                                       layer=1))
+        multiplanar_design.chips['c_chip'] = {
+            'size': {
+                'center_x': '0.0mm',
+                'center_y': '0.0mm',
+                'size_x': '9mm',
+                'size_y': '7mm'
+            }
+        }
+        multiplanar_design.chips['c_chip'] = {
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:539]: duplicate

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 20]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+                'center_x': '0.0mm',
+                'center_y': '0.0mm',
+                'size_x': '9mm',
+                'size_y': '7mm'
+            }
+        }
+        self.assertEqual(
+            BoundsForPathAndPolyTables(multiplanar_design).get_x_y_for_chip(
+                'c_chip'), ((-4.5, -3.5, 4.5, 3.5), 0))
+        self.assertEqual(
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:550]: duplicate

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 21]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+                          "qc1",
+                          options=dict(pos_x="0mm",
+                                       pos_y="-0.08mm",
+                                       pad_width="120um",
+                                       pad_height="30um",
+                                       cpw_stub_height="250um",
+                                       chip="c_chip",
+                                       layer=1))
+        self.assertEqual(
+            LayerStackHandler(
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:625]: Nice

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 22]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:667]: nice

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 23]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+        qc1 = QuadCoupler(multiplanar_design,
+                          "qc1",
+                          options=dict(pos_x="0mm",
+                                       pos_y="-0.08mm",
+                                       pad_width="120um",
+                                       pad_height="30um",
+                                       cpw_stub_height="250um",
+                                       chip="c_chip",
+                                       layer=1))
+        self.assertEqual(
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:747]: nice

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 24]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+            LayerStackHandler(multiplanar_design,
+                              (ls_file_path, None)).get_unique_layer_ints(),
+            {1, 3})
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:775]: nice

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 25]
```diff
@@ -324,10 +348,532 @@ def test_toolbox_metal_cross(self):
...
+                          options=dict(pos_x="0mm",
+                                       pos_y="-0.08mm",
+                                       pad_width="120um",
+                                       pad_height="30um",
+                                       cpw_stub_height="250um",
+                                       chip="c_chip",
+                                       layer=1))
+        self.assertEqual(
+            LayerStackHandler(multiplanar_design,
+                              (ls_file_path, None))._warning_search(
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:824]: nice

File: qiskit_metal/tests/test_toolbox_metal.py [hunk 26]
```diff
@@ -324,10 +348,514 @@ def test_toolbox_metal_cross(self):
...
+                'size_x': '9mm',
+                'size_y': '7mm'
+            }
+        }
+        self.assertEqual(
+            BoundsForPathAndPolyTables(multiplanar_design).get_x_y_for_chip(
+                'c_chip'), ((-4.5, -3.5, 4.5, 3.5), 0))
+        self.assertEqual(
+            BoundsForPathAndPolyTables(multiplanar_design).get_x_y_for_chip(
+                'TEST_chip'), ((), 1))
```
Comment (priti-ashvin-shah-ibm) [inline qiskit_metal/tests/test_toolbox_metal.py:534]: Nice idea for TEST_chip to give error.

**Review by priti-ashvin-shah-ibm — APPROVED** [review]
LGTM

**Pull request merged**